site/guide/state.md

```diff
@@ -85,6 +85,19 @@ fn from_request(req: &'a Request<'r>) -> request::Outcome<T, ()> {
 
 [`Request::guard()`]: https://api.rocket.rs/rocket/struct.Request.html#method.guard
 
+### Request-Local State
+
+In addition to application-wide state, each `Request` has its own state container
+that can be used by request guards and fairings. This container can be used to
+cache expensive computations and lookups, such as to a database or external
+server, that might be used by multiple request guards during routing. Fairings
+can provide request-specific data for consumption by request guards in the same
+manner. See the documentation for the [`FromRequest`] and [`Fairing`] traits for
```

Comment:
I would be more specific about what kind of things you would want to use request-local state for and what problem they're solving. We want users to read this and have a pretty good idea about when to reach for request-local state. I don't think this accomplishes this quite yet.

Don't be afraid to include some example code here too to drive the code home.